packages/contracts-bedrock/src/L1/SystemConfig.sol

```diff
@@ -416,4 +425,19 @@ contract SystemConfig is OwnableUpgradeable, ISemver {
 
         _resourceConfig = _config;
     }
+
+    /// @notice Updates the challenger. Can only be called by the owner.
+    /// @param _challenger New challenger address.
+    function setChallenger(address _challenger) external onlyOwner {
+        _setChallenger(_challenger);
+    }
+
+    /// @notice Internal function for updating the challenger.
+    /// @param _challenger New challenger.
+    function _setChallenger(address _challenger) internal {
+        challenger = _challenger;
+
+        bytes memory data = abi.encode(_challenger);
+        emit ConfigUpdate(VERSION, UpdateType.CHALLENGER, abi.encode(data));
```

Comment:
Remove this `emit ConfigUpdate` otherwise it'll break `op-node`

Reply:
It was always said to be best practice to emit events so that the set of events can be used to reconstruct the state without needing to execute all txs. Then shadow events became a thing where you basically can insert events that don't consume gas or impact state into execution and index them which makes events kind of obsolete if you are running these shadow event indexers

Reply:
I think that including one is the safer option?
Have added it anyways.